cura/Settings/ContainerManager.py

```diff
@@ -350,6 +351,194 @@ def importContainer(self, file_url):
 
         return { "status": "success", "message": "Successfully imported container {0}".format(container.getName()) }
 
+    @pyqtSlot(result = bool)
+    def updateQualityChanges(self):
+        global_stack = UM.Application.getInstance().getGlobalContainerStack()
+
+        containers_to_merge = []
+
+        global_quality_changes = global_stack.findContainer(type = "quality_changes")
+        if not global_quality_changes or global_quality_changes.isReadOnly():
+            UM.Logger.log("e", "Could not update quality of a nonexistant or read only quality profile")
+            return False
+
+        UM.Application.getInstance().getMachineManager().blurSettings.emit()
+
+        containers_to_merge.append((global_quality_changes, global_stack.getTop()))
+
+        for extruder in cura.Settings.ExtruderManager.getInstance().getMachineExtruders(global_stack.getId()):
+            quality_changes = extruder.findContainer(type = "quality_changes")
+            if not quality_changes or quality_changes.isReadOnly():
+                UM.Logger.log("e", "Could not update quality of a nonexistant or read only quality profile")
+                return False
+
+            containers_to_merge.append((quality_changes, extruder.getTop()))
+
+        for merge_into, merge in containers_to_merge:
+            self._performMerge(merge_into, merge)
+
+        UM.Application.getInstance().getMachineManager().activeQualityChanged.emit()
+
+        return True
+
+    @pyqtSlot()
+    def clearUserContainers(self):
+        global_stack = UM.Application.getInstance().getGlobalContainerStack()
+
+        UM.Application.getInstance().getMachineManager().blurSettings.emit()
+
+        for extruder in cura.Settings.ExtruderManager.getInstance().getMachineExtruders(global_stack.getId()):
+            extruder.getTop().clear()
+
+        global_stack.getTop().clear()
+
+    @pyqtSlot(result = bool)
+    def createQualityChanges(self):
+        global_stack = UM.Application.getInstance().getGlobalContainerStack()
+        if not global_stack:
+            return False
+
+        quality_container = global_stack.findContainer(type = "quality")
+        if not quality_container:
+            UM.Logger.log("w", "No quality container found in stack %s, cannot create profile", global_stack.getId())
+            return False
+
+        UM.Application.getInstance().getMachineManager().blurSettings.emit()
+
+        unique_name = UM.Settings.ContainerRegistry.getInstance().uniqueName(quality_container.getName())
+        unique_id = unique_name.lower()
+        unique_id.replace(" ", "_")
+
+        stacks = [ s for s in cura.Settings.ExtruderManager.getInstance().getMachineExtruders(global_stack.getId()) ]
```

Comment:
You could just use `list(...)` to convert a generator into a list, instead of this list comprehension.